packages/cardano-graphql-services/src/DataProjection/Service.ts

```diff
@@ -0,0 +1,55 @@
+import { ChainFollower } from './ChainFollower';
+import { ConnectionConfig } from '@cardano-ogmios/client';
+import { DgraphClient } from './DgraphClient';
+import { MetadataClient } from '../MetadataClient/MetadataClient';
+import { RunnableModule } from '../RunnableModule';
+import { createAssetBlockHandler } from './blockHandlers/AssetBlockHandler';
+import { dummyLogger } from 'ts-log';
+
+export interface ServiceConfig {
+  ogmios?: {
+    connection?: ConnectionConfig;
+  };
+  dgraph: {
+    address: string;
+    schema: string;
+  };
+  metadata: {
+    uri: string;
+  };
+}
+
+export class Service extends RunnableModule {
+  #chainFollower: ChainFollower;
+  #dgraphClient: DgraphClient;
+  #metadataClient: MetadataClient;
+
+  constructor(public config: ServiceConfig, logger = dummyLogger) {
+    super('Service', logger);
+    this.#dgraphClient = new DgraphClient(config.dgraph.address, logger);
+    this.#metadataClient = new MetadataClient(config.metadata.uri);
+    this.#chainFollower = new ChainFollower(
+      this.#dgraphClient,
+      [createAssetBlockHandler(this.#metadataClient, logger)],
+      logger
+    );
+  }
+
+  async initialize() {
+    super.initializeBefore();
+    await this.#dgraphClient.initialize(this.config.dgraph.schema);
+    await this.#chainFollower.initialize(this.config.ogmios?.connection);
+    await this.#metadataClient.initialize();
+    super.initializeAfter();
+  }
+
+  async startImpl() {
+    // Todo: Get most recent point to start sync from
```

Comment:
We should probably implement a minimal block handler that will write block data so that we can query the tip here

Reply:
Yes, see my comment here from a pending review I forgot to submit. 
https://github.com/input-output-hk/cardano-js-sdk/pull/197#discussion_r830489411